File: src/main/java/org/sonar/plugins/xquery/language/SourceCode.java
/*
 * © 2014 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.language;

import org.sonar.api.design.Dependency;
import org.sonar.api.measures.Measure;
import org.sonar.api.measures.Metric;
import org.sonar.api.resources.File;

import java.util.List;

public interface SourceCode {

    public String getCodeString();

    public List<String> getCode();

    public List<Issue> getIssues();

    public List<Dependency> getDependencies();

    public List<Measure> getMeasures();

    public void addIssue(Issue issue);

    public void addMeasure(Metric metric, double value);

    public void addDependency(File dependencyResource);

    public File getResource();

    public Measure getMeasure(Metric metric);

}

File: src/main/java/org/sonar/plugins/xquery/language/SourceCode.kt
/*
 * © 2014 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.language

import org.sonar.api.design.Dependency
import org.sonar.api.measures.Measure
import org.sonar.api.measures.Metric
import org.sonar.api.resources.File
import java.io.Serializable

interface SourceCode {

    val codeString: String

    val code: List<String>

    val issues: List<Issue>

    val dependencies: List<Dependency>

    val measures: List<Measure<*>>

    val resource: File

    fun addIssue(issue: Issue)

    fun <T : Serializable> addMeasure(metric: Metric<T>, value: Double)

    fun addDependency(dependencyResource: File)

    fun <T : Serializable> getMeasure(metric: Metric<T>): Measure<*>?

}
